imagetest/test_suites/network/alias_ip_test.go

```diff
@@ -0,0 +1,98 @@
+package network
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/guest-test-infra/imagetest/utils"
+)
+
+const (
+	markerFile = "/boot-marker"
+)
+
+func TestAliasAfterReboot(t *testing.T) {
+	var networkInterface string
+
+	image, err := utils.GetMetadata("image")
+	if err != nil {
+		t.Fatalf("couldn't get image from metadata")
+	}
+
+	switch {
+	case strings.Contains(image, "debian-10") || strings.Contains(image, "ubuntu"):
+		networkInterface = defaultPredictableInterface
+	default:
+		networkInterface = defaultInterface
+	}
+
+	_, err = os.Stat(markerFile)
+	if os.IsNotExist(err) {
+		// first boot
+		if _, err := os.Create(markerFile); err != nil {
+			t.Fatalf("failed creating marker file: %v", err)
+		}
+		t.Fatalf("failed on first boot")
+	}
+	// second boot
+	actual, err := getAliasIP(networkInterface)
+	if err != nil {
+		t.Fatal(err)
+	}
+	expected, err := utils.GetMetadata("network-interfaces/0/ip-aliases/0/")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if string(actual) != string(expected) {
+		t.Fatalf("alias ip is not as expected after reboot, expected %s, actual %s", expected, actual)
+	}
+}
+
+func getAliasIP(networkInterface string) ([]byte, error) {
```

Comment:
on reflection this function is more general than alias IPs, can be named 'getGoogleRoutes'